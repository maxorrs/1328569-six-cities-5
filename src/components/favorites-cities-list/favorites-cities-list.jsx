import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import PlacesListFavorite from '../places-list-favorite/places-list-favorite';
import {getFavoritesAdaptSelector, getSelectedCitySelector, getUniqueCitiesSelector} from '../../store/reducers/data/selectors';
import {getFavoritesByCity} from '../../utils/common';

const CitiesList = ({uniqueCities, favorites, selectedCity}) => {
  return (
    <ul className="favorites__list">
      {
        uniqueCities.map((city) => {
          const activeClass = city === selectedCity ? `locations--current` : ``;
          const favoritesByCity = getFavoritesByCity(favorites, city);
          return (
            <li
              key={city}
              className="favorites__locations-items">
              <div className={`favorites__locations locations ${activeClass}`}>
                <div className="locations__item">
                  <a href="#" className="locations__item-link">
                    <span>{city}</span>
                  </a>
                </div>
              </div>
              <div className="favorites__places">
                <PlacesListFavorite
                  favorites={favoritesByCity} />
              </div>
            </li>
          );
        })
      }
    </ul>
  );
};

CitiesList.propTypes = {
  uniqueCities: PropTypes.array.isRequired,
  favorites: PropTypes.array.isRequired,
  selectedCity: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  uniqueCities: getUniqueCitiesSelector(state),
  selectedCity: getSelectedCitySelector(state),
  favorites: getFavoritesAdaptSelector(state)
});

export {CitiesList};
export default connect(mapStateToProps)(CitiesList);
