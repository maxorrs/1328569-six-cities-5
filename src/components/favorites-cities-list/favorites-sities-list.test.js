import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

import {FavoritesCitiesList} from './favorites-cities-list';

const getOffersMock = (count) => {
  const templateOffers = Array(count)
    .fill(``)
    .map((_, index) => {
      return {
        'city': {
          'name': `Amsterdam`,
          'location': {
            'latitude': 51.225402,
            'longitude': 6.776314,
            'zoom': 13
          }
        },
        'preview_image': `url`,
        'images': [`url1`, `url2`],
        'title': `Canal View Prinsengracht`,
        'is_favorite': Boolean(Math.random()),
        'is_premium': false,
        'rating': 4.7,
        'type': `room`,
        'bedrooms': 1,
        'max_adults': 3,
        'price': 298,
        'goods': [`Laptop friendly workspace`, `Breakfast`, `Washer`],
        'host': {
          'id': 25,
          'name': `Angelina`,
          'is_pro': true,
          'avatar_url': `img/avatar-angelina.jpg`
        },
        'description': `Description`,
        'location': {
          'latitude': 51.222402,
          'longitude': 6.786314,
          'zoom': 16
        },
        'id': index + 1
      };
    });

  if (count === 1) {
    const [oneOffer] = templateOffers;
    return oneOffer;
  }

  return templateOffers;
};

const offersMock = getOffersMock(3);

it(`FavoritesCitiesList is rendered correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <FavoritesCitiesList
            uniqueCities={[`Amsterdam`, `Paris`]}
            favorites={offersMock}
            selectedCity={`Amsterdam`} />
        </BrowserRouter>
    )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
