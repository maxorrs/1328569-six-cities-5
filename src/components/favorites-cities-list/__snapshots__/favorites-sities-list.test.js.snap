// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesCitiesList is rendered correctly 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="place-card favorites__place-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="110"
              src="url"
              width="150"
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                298
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "94%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/1"
              onClick={[Function]}
            >
              Canal View Prinsengracht
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__place-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="110"
              src="url"
              width="150"
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                298
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "94%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/2"
              onClick={[Function]}
            >
              Canal View Prinsengracht
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__place-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="110"
              src="url"
              width="150"
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                298
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "94%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/3"
              onClick={[Function]}
            >
              Canal View Prinsengracht
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
    </div>
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations "
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Paris
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    />
  </li>
</ul>
`;
